File: cs50-projects/sentimental-readability/readability.py
from string import ascii_lowercase


def main():
    # Read a string from user
    s = input("Enter a string: ")
    # Number of letters
    letters = get_letters(s)
    # Number of words
    words = get_words(s)
    # Number of sentances
    sentences = get_sentences(s)
    # Coleman-Liau index (CLI)
    index = cli(letters, words, sentences)

    if index <= 1:
        # if indes is less than 1 we output "Before Grade 1"
        print("Before Grade 1")
    elif index >= 16:
        # if index is 16 or higher we output "Grade 16+"
        print("Grade 16+")
    else:
        # if grade is less than 16 and higer than 1 we output "Grade "+ index
        print(f"Grade {index}")


def get_letters(s):
    # if element is in alfabet then is a letter
    alfabet = list(ascii_lowercase)
    count = 0
    for letter in s:
        if letter.lower() in alfabet:
            count = count + 1

    return int(count)


def get_words(s):
    # we create a list with all words in the string then return lenght of list
    word_list = s.split()
    return int(len(word_list))


def get_sentences(s):
    # we check for end of a sentence ("!", ".", "?")
    count = 0
    for letter in s:
        if letter in ["!", ".", "?"]:
            count = count + 1

    return int(count)


def cli(letters, words, sentences):
    # calculates Coleman-Liau index (CLI)
    l = letters / words * 100
    s = sentences / words * 100
    return int(round(0.0588 * l - 0.296 * s - 15.8))


main()

File: cs50-projects/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int get_letters(string s);
int get_words(string s);
int get_sentences(string s);
int cli(int letters, int words, int sentances);


int main(void)
{
    //Read a string from user
    string s = get_string("Enter a string: ");
    //number of letters
    int letters = get_letters(s);
    //number of words
    int words = get_words(s);
    //number of sentances
    int sentances = get_sentences(s);
    //Coleman-Liau index (CLI)
    int index = cli(letters, words, sentances);


    if (index <= 1)
    {
        //if indes is less than 1 we output "Before Grade 1"
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        //if index is 16 or higher we output "Grade 16+"
        printf("Grade 16+\n");

    }
    else
    {
        //If grade is less than 16 and higer than 1 we output "Grade "+ index
        printf("Grade %i\n", index);
    }
}


int get_letters(string s)
{
    // counts the number of letters
    int l = strlen(s);
    int i, cont = 0;
    for (i = 0; i < l ; i++)
    {
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            cont++;

        }
    }
    return cont;
}
int get_words(string s)
{
    // counts the number of words
    int l = strlen(s);
    int i, cont = 1;
    for (i = 0 ; i < l; i++)
    {
        if (s[i] == ' ')
        {
            cont++;

        }
    }
    return cont;
}

int get_sentences(string s)
{
    // counts the number of sentances
    int l = strlen(s);
    int i, cont = 0;
    for (i = 0 ; i < l; i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            cont++;

        }
    }
    return cont;
}

int cli(int letters, int words, int sentances)
{
    //calculates Coleman-Liau index (CLI)
    float L = letters / (float) words * 100;
    float S = sentances / (float) words * 100;
    int index = (int) round(0.0588 * L - 0.296 * S - 15.8);
    return index;

}